src/main/java/solutions/siren/join/index/query/FieldDataTermsQueryHelper.java

```diff
@@ -73,28 +52,73 @@ public final static long hash(BytesRef term) {
   /**
    * Encodes a long into the byte array dst at the given offset.
    */
-  public final static void writeInt(byte[] dst, int offset, int i) {
-    dst[offset] = ((byte) (i >> 24));
-    dst[offset + 1] = ((byte) (i >> 16));
-    dst[offset + 2] = ((byte) (i >> 8));
-    dst[offset + 3] = ((byte) i);
+  public final static void writeInt(BytesRef dst, int i) {
```

Comment:
I don't how many times this is called, but revert to previous `writeInt` and add `dst.offset += 4;` at the end ?